emmet-core/emmet/core/qchem/calculation.py

```diff
@@ -0,0 +1,437 @@
+"""Core definitions of a QChem calculation document."""
+
+# mypy: ignore-errors
+
+import logging
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+from pydantic import BaseModel, Field
+from pydantic.datetime_parse import datetime
+from pymatgen.io.qchem.inputs import QCInput
+from pymatgen.io.qchem.outputs import QCOutput
+from collections import OrderedDict
+import re
+
+from emmet.core.qchem.calc_types import (
+    LevelOfTheory,
+    CalcType,
+    TaskType,
+    level_of_theory,
+    calc_type,
+    task_type,
+)
+
+from emmet.core.qchem.task import QChemStatus
+
+__author__ = "Rishabh D. Guha <rdguha@lbl.gov>"
+logger = logging.getLogger(__name__)
+
+# class QChemObject(ValueEnum):
+# Not sure but can we have something like GRAD and HESS
+# as QChem data objects
+
+
+class CalculationInput(BaseModel):
+    """
+    Document defining QChem calculation inputs.
+    """
+
+    initial_molecule: Dict[str, Any] = Field(
+        None, description="input molecule geometry before the QChem calculation"
+    )
+
+    # parameters: Dict[str, Any] = Field(
+    #     None, description = "Parameters from a previous QChem calculation."
+    # )
+
+    charge: int = Field(None, description="The charge of the input molecule")
```

Comment:
If charge is top-level, spin multiplicity should be as well

Reply:
Agree in principle, but if we're just duplicating atomate1's [`generate_doc`](https://github.com/hackingmaterials/atomate/blob/6327b07ea1ae85fbbe29772777a80dfe4a94dc7b/atomate/qchem/drones.py#L146) method, only `molecule` is stored at the top level. That method overrides the molecule charge, too.

Would it be enough to store an entire `Molecule` object under `initial_molecule`, that will carry charge and spin information with it?